client/app/components/Modal/index.jsx

```diff
@@ -97,23 +98,53 @@ export class Modal extends React.Component<Props, State> {
     this.setState({ open: !open });
   };
 
-  render() {
+  resolveElement = () => {
     const { element, elementId } = this.props;
+    let Component = null;
+    let renderComponent = null;
+
+    if (typeof element === 'object' && element.component) {
+      const { component, props } = element;
+      Component = this.resolveComponent(component);
+      renderComponent = <Component {...props} />;
```

Comment:
You could alternatively use `React.createElement()` to create your Avatar component and pass in the props.